File: 01_Two_Sum.py
class Solution:
    def twoSum(self, nums, target):
        output = [0, 1]
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    output[0] = i
                    output[1] = j
                    break
        return output
    
# Time Complexity: O(n^2)
# Space Complexity: O(1)

File: C#/01_Two_Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> dict = new Dictionary<int,int>();
        int diff=0;
        for(int i=0; i<nums.Length; i++)
        {
            diff = target- nums[i];
            if(dict.ContainsKey(diff))
            {
               return new int[]{i,dict[diff]};
            }
            else if(!dict.ContainsKey(nums[i]))
            {
              dict.Add(nums[i],i);
            }
        }
        return null;
    } 
}
    
# Time Complexity: O(n)
# Space Complexity: O(1)
